pkg/reconciler/instances/ory/test/ory_integration_test.go

```diff
@@ -0,0 +1,63 @@
+package test
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/reconciler/pkg/logger"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"github.com/kyma-incubator/reconciler/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+const (
+	envOryIntegrationTests = "ORY_RECONCILER_INTEGRATION_TESTS"
+	namespace              = "kyma-system"
+)
+
+func OryIntegrationTest(t *testing.T) {
+	if !isIntegrationTestEnabled() {
+		t.Skipf("Integration tests disabled: skipping parts of test case '%s'", t.Name())
+	}
+}
+
+func isIntegrationTestEnabled() bool {
+	expensiveTests, ok := os.LookupEnv(envOryIntegrationTests)
+	return ok && (expensiveTests == "1" || strings.ToLower(expensiveTests) == "true")
+}
+
+func TestOryIntegration(t *testing.T) {
+	OryIntegrationTest(t)
+
+	log := logger.NewLogger(false)
```

Comment:
I can see a pattern here. Each test suite would contain the same set of properties such as: logger, kubeClient, context and metadata. What about preparing this in a separate function? For now the initialisation part is longer than a test logic itself so we could move this boilerplate elsewhere and refuse later. Maybe an introduction of a helper struct could become handy here so whenever we create the struct it will be created from scratch to not be reused between test cases?

Reply:
Let's start with this structure, indeed.